webapi/src/main/scala/org/knora/webapi/messages/util/search/gravsearch/GravsearchParser.scala

```diff
@@ -112,22 +112,12 @@ object GravsearchParser {
        * @param sourceName the source name.
        * @return an [[algebra.Var]] representing the name or its literal value.
        */
-      def nameToVar(sourceName: String): algebra.Var = {
-        val sparqlVar = new algebra.Var
-        sparqlVar.setName(sourceName)
-
+      def nameToVar(sourceName: String): algebra.Var =
         valueConstants.get(sourceName) match {
-          case Some(valueConstant) =>
-            sparqlVar.setConstant(true)
-            sparqlVar.setValue(valueConstant.getValue)
-
-          case None =>
-            sparqlVar.setConstant(false)
+          case Some(valueConstant) => new algebra.Var(sourceName, valueConstant.getValue(), false, true)
+          case None                => new algebra.Var(sourceName, null, false, false)
```

Comment:
Could `null` be replaced by `Option` here?

Reply:
No, I don't think so (or at least I wouldn't know how) as this comes from a library and a specific object is needed as parameter (which we don't have at that place).
This "fix" removes one of the deprecation warnings we're currently having.